bokeh/colors/util.py

```diff
@@ -119,12 +119,40 @@ def __init__(self, name: str, r: int, g: int, b: int) -> None:
         self.name = name
         super().__init__(r, g, b)
 
+    @classmethod
+    def find(cls, name: str) -> NamedColor:
+        # If name not recognised, raises ValueError
+        index = cls.__all__.index(name)
+        return cls.colors[index]
+
     def to_css(self) -> str:
         '''
 
         '''
         return self.name
 
+def _color_as_rgb_hex(color: str) -> str:
+    ''' Convert string color to RGB hex string.
+
+    Color may be a named color such as "blue", or a hex RGB(A) string such as
+    "#FFAA00", "#FFAA00FF", "#FA0" or "#FA0F". Returned string is always of
+    the form "#RRGGBB" with 2 characters for each of the R, G and B components
+    and no alpha component.
+
+    '''
+
+    if color.startswith("#"):
+        if len(color) in (7, 9):
+            return color[:7].upper()
+        elif len(color) in (4, 5):
+            return f"#{color[1]*2}{color[2]*2}{color[3]*2}".upper()
+        else:
+            raise ValueError(f"Unable to parse hex color string: '{color}'")
+    try:
+        return NamedColor.find(color).to_hex()
+    except ValueError:
+        raise ValueError(f"Color '{color}' must be either a named color or a hex RGB(A) string")
+
```

Comment:
This seems like a useful function, but silently dropping the alpha seems like confusing behavior that limits how useful and safe this function is. Why not keep the alpha here in this function (forcing it to FF if not present), but then explicitly strip it in `single_color_palette`? Stripping it seems easy once it's in a canonical format, which in this case would include the alpha always.

Reply:
I don't have a strong opinion either way. At the moment the function is only used by `varying_alpha_palette()` and is called `.. as_rgb_hex` to explicitly indicate alpha is not part of the return. But I don't object to it including alpha in the return and the name changing to `.. as_rgba_hex`.